021-server-side-rendering/021-server-side-rendering.md

```diff
@@ -0,0 +1,398 @@
+
+```js
+import { Client, Account } from "node-appwrite";
+
+const client = new Client();
+
+client.setEndpoint("https://cloud.appwrite.io/v1");
+client.setProject("PROJECT_ID");
+client.setKey("API_KEY");
+client.setSession("SESSION_SECRET");
+```
+
+### Issue #5: Web SDK File Uploads
+
+The Web SDK only accepts a `File` object when uploading files. This is not compatible with SSR frameworks, as the `File` object is only available in the browser.
+
+There is a workaround for this, which involves importing the `File` polyfill from the `formdata-node` package, and converting the blob to an array buffer.
+
+```js
+import fetch from 'node-fetch';
+import { FormData, File } from 'formdata-node';
+
+export const uploadFile = async (blob: Blob) => {
+	const path = `/storage/buckets/${BUCKET_ID}/files`;
+	const uri = new URL(ENDPOINT + path);
+
+	const formData = new FormData();
+	const file = new File([await blob.arrayBuffer()], blob.name);
+	formData.set('file', file);
+	formData.set('fileId', ID.unique());
+
+	const res = await fetch(uri.toString(), {
+		method: 'post',
+		headers: {
+			...client.headers,
+			'x-appwrite-project': PROJECT_ID,
+		},
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		body: formData as any,
+	});
+
+	console.log(res);
+
+	return res;
+};
+```
+
+#### Proposed Solution
+
+Match the existing Node SDK which accepts an InputFile helper object.
+
+```js
+storage.createFile(
+  "[BUCKET_ID]",
+  "[FILE_ID]",
+  InputFile.fromBlob(blob, "file.jpg")
+);
+```
+
+### Documentation & Content
+
+#### What **docs** would support this feature?
+
+- SDK docs and examples for the new helper method.
+- SDK docs and examples for the new token exchange endpoint.
+
+#### What **tutorials** (text/video) might help developers understand this feature scope, capabilities, and possible use-cases?
+
+- Generic Tutorial for using Appwrite for SSR
+- Tutorial for using Appwrite and Next.js for SSR
+- Tutorial for using Appwrite and Nuxt.js for SSR
+- Tutorial for using Appwrite and Remix for SSR
+- Tutorial for using Appwrite and SvelteKit for SSR
+
+#### What **demo applications** can help us demonstrate this feature APIs and capabilities?
+
+- Update Almost SSR examples.
+
+### Prior art
+
+[prior-art]: #prior-art
+
+- https://supabase.com/docs/guides/auth/server-side-rendering
+- https://firebase.nuxtjs.org/tutorials/ssr.html
+- https://github.com/gladly-team/next-firebase-auth
+
+### Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+**Problem #2**
+
+2.1. Should we remove the undocumented OAuth2 SSR workaround?
+
+2.2. Do we need to add a new endpoint to exchange the temporary token for a session token? Can we reuse the existing magic URL exchange endpoint?
+
+**General**
+
+4.1. Some of these changes reduce the distinction between the Server and Client Side SDKs. Which SDKs would be recommended for SSR applications?
```

Comment:
Define something that we can clearly explain and suggest in docs. 

If it feels complex to explain, probably too complicated.

Reply:
If you want to make privileged actions, such as provisioning databases and collections, you need to use the server API and authenticate via API key.

If you want to act as a single user, you need to authenticate using a session from your cookie jar, or set using the setSession helper.

Reply:
@loks0n sounds good to me.